openghg/transform/flux/_edgar.py

```diff
@@ -587,95 +565,69 @@ def _extract_file_info(edgar_file: Union[Path, str]) -> Dict:
+    cleaning_pat = re.compile(rf"({cleaning_regexps_joined})")
+
+    filename_clean = cleaning_pat.sub("", filename)
+
+    # parse string of form:
+    # {version}_{species}_{optional co2 info}_{year}_{optional month}_{optional sector}_{optional resolution}
+    info_pat = re.compile(
+        r"^(?P<version>v\d[\.\d]*)_"  # capture version e.g. v432, v50, v8.0
+        r"(?P<species>[a-zA-Z\d-]+)_"  # capture species, e.g. CH4, c-C4F8, HFC-43-10-mee
+        r"((?P<co2_options>(excl|org)_short-cycle)(_org)?(_C)?_)?"  # optional, capture CO2 info
+        r"(?P<year>(19|20)\d{2})"  # capture year (might not end in _)
+        r"(_(?P<month>\d{1,2}))?"  # optionally capture month
+        r"(_(?P<sector>(\w+)))?"  # optionally capture sector, e.g. TOTALS, TNR_Ship, N2O, IPCC_4C_4D1_4D4
+        r"(\.(?P<resolution>\d\.\dx\d\.\d))?"  # optionally capture resolution, e.g. 0.1x0.1
+    )
 
-    # Check if year can be cast to integer to check this is a valid value
-    try:
-        year_str = filename_split[index_remaining]
-        year = int(year_str)
-    except IndexError:
-        raise ValueError(f"Unable to cast year extracted from file format to an integer: {year_str}")
-    except ValueError:
-        # In some files there is no source specified so
-        # filename_split[2] contains the year and resolution
-        # e.g. "v50_CH4_2015.0.1x0.1.nc" --> "2015.0.1x0.1"
-        try:
-            year = int(year_str.split(".")[0])
-        except ValueError:
-            raise ValueError(f"Could not find valid year value from file: {filename}")
+    if m := info_pat.search(filename_clean):
+        file_info = m.groupdict()
     else:
-        index_remaining += 1
+        # info_pat matches all files in known EDGAR versions
+        # (verified by searching all file names for these versions
+        # 18 March 2024)
+        raise ValueError(f"Did not recognise input file format: {filename}")
 
-    # Check whether month is included in filename
-    # e.g. "v6.0_CH4_2015_1_ENE.0.1x0.1.nc"
-    try:
-        month: Optional[int] = int(filename_split[3])
-    except (IndexError, ValueError):
-        month = None
-    else:
-        index_remaining += 1
+    # make "source" string
+    co2_options = file_info.pop("co2_options") or ""
 
-    # Attempt to extract source(s) and resolution from filename stem
-    # e.g. "v6.0_CH4_2015_TOTALS.0.1x0.1.nc" --> "TOTALS.0.1x0.1"
-    # e.g. "v50_CH4_2015.0.1x0.1.nc" --> "2015.0.1x0.1" (note no source in filename)
-    # e.g. "v432_CH4_2010_9_IPCC_6A_6D.0.1x0.1.nc" --> "IPCC_6A_6D.0.1x0.1"
-    try:
-        source_resolution = "-".join(filename_split[index_remaining:])
-    except (IndexError, ValueError):
-        raise ValueError(f"Unable to extract source: {filename}")
+    sector = file_info.pop("sector") or ""
+    sector = sector.replace("_", "-")
+
+    source = "_".join([co2_options, sector])
+    source = source.strip("_")  # in case co2_options or sector is ""
```

Comment:
@rt17603 this results in slightly different source names than the old code in the case where there is a CO2 specific string like "org_short-cycles_C" but no sector string like "TOTALS". The old code would make the source "org_short-cycles_" and the new code would make it "org_short-cycles". Is this okay>

Reply:
Yes that seems fine to me.